File: 1-easy/problems/1475-final-prices-with-a-special-discount-in-a-shop/solution.ts
function finalPrices(prices: any): boolean | number | string | any {
  """
  :type prices: List[int]
  :rtype: List[int]
  """
  stk = []
  for (i, p in enumerate(prices)) {
      while (stk && prices[stk[-1]] >= p) {
          prices[stk.pop()] -= p
      stk.append(i)
  return prices;
}

export { finalPrices };

File: 1-easy/problems/1475-final-prices-with-a-special-discount-in-a-shop/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def finalPrices(self, prices):
        """
        :type prices: List[int]
        :rtype: List[int]
        """
        stk = []
        for i, p in enumerate(prices):
            while stk and prices[stk[-1]] >= p:
                prices[stk.pop()] -= p
            stk.append(i)
        return prices
